contrib/test/roles/irc_channel_notice/tasks/main.yml

```diff
@@ -0,0 +1,45 @@
+---
+
+- block:  # delegate_to: localhost b/c 'is_file()', 'csvfile' and 'template' lookups
+
+  - name: All role assertions pass
+    assert:
+      that:  # All are defined and non-empty
+        - 'irc_server | default()'
+        - 'irc_channel | default()'
+        - 'irc_from | default()'
+        - 'irc_to | default() or commit_author | default()'
+        - 'irc_msg_template | default()'
+        - 'irc_nick_xlation | default()'
+
+  - name: Relative paths are made absolute
+    set_fact:
+      _irc_msg_template: '{{ role_path ~ "/templates/" ~ irc_msg_template }}'
+      _irc_nick_xlation: '{{ role_path ~ "/files/" ~ irc_nick_xlation }}'
+
+  - name: Message recipient name is translated
+    set_fact:
+      irc_to: >
+        {{ lookup("csvfile",
+                  commit_author ~
+                  " file=" ~ _irc_nick_xlation ~
+                  " col=1 delimiter=: default=" ~ irc_to) }}
+    when: commit_author | default() and _irc_nick_xlation | is_file
```

Comment:
...the above will do the translation if possible.  Otherwise just use ``irc_to``.